### PR TITLE
CNF-9796: Add functions to cleanup backups and remove OADP 

### PR DESCRIPTION
**Pull request by Missxiaoguo**
Some code refactors and add functions for backups and OADP operator removal which is needed during abort, rollback or finalize.

### CODE REVIEW

File: internal/backuprestore/restore_test.go [hunk 1]
```diff
@@ -414,3 +418,129 @@ func TestLoadRestoresFromDir(t *testing.T) {
...
+	}()
+
+	// Mock the backup storage
+	time.Sleep(1 * time.Second)
+	bsl := fakeBackupStorageBackendWithStatus("oadp-1", velerov1.BackupStorageLocationPhaseAvailable)
+	if err := fakeClient.Create(context.Background(), bsl); err != nil {
+		t.Errorf("error in creating backup storage location")
+	}
+
+	// Verify dpa is created
```
Comment (sudomakeinstall2) [inline internal/backuprestore/restore_test.go:474]: is this a typo?

Comment (Missxiaoguo) [reply internal/backuprestore/restore_test.go:487]: Do you mean "dpa"? Not a typo. It's DataProtectionApplication

**Comment by Missxiaoguo** [comment]
Rebased

**Pull request merged**